### PR TITLE
Add is method to fundamental types

### LINKED ISSUE #860: Add is methods to fundamental types
See https://github.com/gtk-rs/gtk-rs-core/issues/502

### PR DESCRIPTION
**Pull request by bilelmoussaoui**
Fixes #860

### CODE REVIEW

File: gtk4/src/expression.rs [hunk 1]
```diff
@@ -23,9 +23,13 @@ pub unsafe trait IsExpression:
 }
 
 impl Expression {
+    pub fn is<E: IsExpression>(&self) -> bool {
+        self.type_() == E::static_type()
```
Comment (bilelmoussaoui) [inline gtk4/src/expression.rs:27]: I am wondering whether it makes sense to keep the `type_()` method as public

Comment (bilelmoussaoui) [reply gtk4/src/expression.rs:27]: I did that in a separate commit so we can only cherry-pick the first one for the next release

Comment (sdroege) [reply gtk4/src/expression.rs]: Seems useful to get the type?

**Review by sdroege — APPROVED** [review]

**Review by sdroege — CHANGES_REQUESTED** [review]
I would keep `type_()` public.

File: gtk4/src/expression.rs (hunk 1 above)
Comment (bilelmoussaoui) [reply gtk4/src/expression.rs]: I can't see where you would need that in general

Comment (sdroege) [reply gtk4/src/expression.rs]: It's also not hurting to keep it :) It's potentially useful information and you can't get that any other way right now.

People are also using the `type_()` function on objects.

Comment (bilelmoussaoui) [reply gtk4/src/expression.rs]: I dropped that commit, I will add similar functions to Gsk.RenderNode/Gdk.Event then

**Review by sdroege — APPROVED** [review]

**Pull request merged**